app/src/main/java/com/google/samples/apps/sunflower/compose/plantdetail/PlantDetailView.kt

```diff
@@ -0,0 +1,421 @@
+) {
+    Surface(modifier.systemBarsPadding(top = true)) {
+        TopAppBar(
+            backgroundColor = MaterialTheme.colors.surface
+        ) {
+            IconButton(onBackClicked, Modifier.gravity(Alignment.CenterVertically)) {
+                Icon(Icons.Filled.ArrowBack)
+            }
+            Text(
+                text = plantName,
+                style = MaterialTheme.typography.h6,
+                // As title in TopAppBar has extra inset on the left, need to do this: b/158829169
+                modifier = Modifier.weight(1f).fillMaxSize()
+                    .wrapContentSize(Alignment.Center)
+            )
+            val shareAccessibilityLabel = stringResource(R.string.menu_item_share_plant)
+            IconButton(
+                onShareClicked,
+                Modifier.gravity(Alignment.CenterVertically).semantics {
+                    accessibilityLabel = shareAccessibilityLabel
+                }
+            ) {
+                Icon(Icons.Filled.Share)
+            }
+        }
+    }
+}
+
+@Composable
+private fun PlantImageHeader(
+    scrollerPosition: ScrollerPosition,
+    imageUrl: String,
+    onFabClicked: () -> Unit,
+    isPlanted: Boolean,
+    modifier: Modifier = Modifier,
+    transitionModifier: Modifier = Modifier
+) {
+    var imageHeight by state { 0 }
+
+    Stack(modifier.fillMaxWidth()) {
+        PlantImage(scrollerPosition, imageUrl, transitionModifier.onPositioned {
+            imageHeight = it.size.height
+        })
+        if (!isPlanted) {
+            val fabModifier = if (imageHeight != 0) {
+                Modifier
+                    .gravity(Alignment.TopEnd)
+                    .padding(end = 8.dp)
+                    .offset(y = getFabOffset(imageHeight, scrollerPosition))
+                    .plus(transitionModifier)
+            } else {
+                Modifier
+            }
+            val fabAccessibilityLabel = stringResource(R.string.add_plant)
+            FloatingActionButton(
+                onClick = onFabClicked,
+                shape = MaterialTheme.shapes.small,
+                modifier = fabModifier.semantics {
+                    accessibilityLabel = fabAccessibilityLabel
+                }
+            ) {
+                Icon(Icons.Filled.Add)
+            }
+        }
+    }
+}
+
+@Composable
+private fun PlantImage(
+    scrollerPosition: ScrollerPosition,
+    imageUrl: String,
+    modifier: Modifier = Modifier,
+    placeholderColor: Color = MaterialTheme.colors.onSurface.copy(0.2f)
+) {
+    val parallaxOffset = with(DensityAmbient.current) {
+        scrollerParallaxOffset(scrollerPosition)
+    }
+    CoilImageWithCrossfade(
+        data = imageUrl,
+        contentScale = ContentScale.Crop,
+        loading = {
+            Box(modifier = Modifier.fillMaxSize(), backgroundColor = placeholderColor)
+        },
+        modifier = modifier.fillMaxWidth().padding(top = parallaxOffset).preferredHeight(278.dp)
```

Comment:
nit: one modifier per line

Reply:
But... this fits into one line 😱  I really prefer that rather than

```
CoilImageWithCrossfade(
        ...,
        modifier = modifier
            .fillMaxWidth()
            .padding(top = parallaxOffset)
            .preferredHeight(278.dp)
```

Indentation looks... very weird

Reply:
Another example where the indentation doesn't help with readability 

```
val iconModifier = Modifier
    .sizeIn(maxWidth = 32.dp, maxHeight = 32.dp)
    .drawBackground(color = MaterialTheme.colors.surface, shape = CircleShape)
```

Reply:
I do like the conciseness, but I think this is probably a good rule for us to keep. When reading this I missed preferredHeight the first time because it was too far from the start of the line :).